Add support for new gamma (auth.chalmers.it)

ViddeM opened the pull request.

ViddeM committed:
Began supporting gamma 2

ViddeM committed:
Add TODos

ViddeM committed:
Keep working it

ViddeM committed:
Implement updated auth

ViddeM committed:
Updated docker-compose vars

ViddeM committed:
Fix clippy warnings

ViddeM committed:
Update default auth url to use https

ViddeM committed:
Remove extra log

ViddeM committed:
Fix import quotations

ViddeM commented on backend/src/repositories/user.rs:
```diff
@@ -8,32 +9,54 @@ use crate::{
+  }
+
+  pub async fn get_by_cid(&self, cid: &str) -> HubbitResult<GammaUser> {
+    log::warn!(
```
This is only used as a fallback and I'm not sure if it ever occurs in a real-world use-case (unless someone manually removes it from the cache).

ViddeM committed:
Remove old gamma from dev flow

ViddeM committed:
Fix build flow

Oscariremma commented on backend/Cargo.toml:
```diff
@@ -43,3 +44,4 @@ tokio = { version = "1.36.0", features = [
     "time",
 ] }
 uuid = { version = "1.7.0", features = ["serde"] }
+gamma_rust_client = { git = "https://github.com/viddem/gamma.git", rev = "1235842562c9f3b50a68582dea13f66fbe93cf5e" }
```
Small TODO here. We should switch to the cthit hosted gamma repo and not a fork when the [gamma pr](https://github.com/cthit/Gamma/pull/880) is merged

ViddeM replied:
Done!

Oscariremma reviewed (approved):
LGTM

ViddeM committed:
Reset to 13.2

Prod is using 13.2 still so better to match that (simplifies importation of backups etc).

ViddeM committed:
Update to using cthit version of gamma client
